pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -456,12 +455,6 @@ func (r *ReconcileLogStorage) Reconcile(ctx context.Context, request reconcile.R
 			return reconcile.Result{}, err
 		}
 
-		applyTrial, err = r.shouldApplyElasticTrialSecret(ctx)
-		if err != nil {
-			r.status.SetDegraded("Failed to get eck trial license", err.Error())
-			return reconcile.Result{}, err
-		}
-
 		if esLicenseType, err = utils.GetElasticLicenseType(ctx, r.client, reqLogger); err != nil {
```

Comment:
Basically we still need this function for clusters that are upgraded from a lower version within a month of installing the old version, right?

Is it possible to just remove it and always use the 'basic auth' login trick for kibana? That would make the possibly upcoming changes a lot easier. If you prefer to defer this one more release, I can understand.

Reply:
There is no plan to remove that check yet, there might be customers with paid Elasticsearch license, if so they will continue to use Kibana's OIDC.